core/src/main/java/org/ldaptive/LdapEntry.java

```diff
@@ -59,8 +63,8 @@ public class LdapEntry extends AbstractMessage
   /** Normalized LDAP DN. */
   private String normalizedDn;
 
-  /** LDAP attributes on the entry. */
-  private Map<String, LdapAttribute> attributes = new LinkedHashMap<>();
+  /** Whether this object has been marked immutable. */
+  private volatile boolean immutable;
```

Comment:
Why volatile?

Reply:
Perhaps overzealous thinking on my part to make data consistency safe in multi-threaded environments. You could convince me this isn't needed since other properties aren't thread safe. But maybe this flag should be given special consideration.

Reply:
Given the key importance of the flag in implementing immutability, I concede it may merit special treatment w/r/t under concurrency. That said, more analysis and test coverage in the future may be warranted.